### ISSUE TITLE
Сustomizing resources generated by Chalice (and cdk-chalice respectively)

### ISSUE THREAD

**Issue by natCyberark** [alexpulver/cdk-chalice#15]
Hi !

We have a problem to enable API-Gateway logging in our cdk cloudformation template and providing it with role, that allows writing to cloudwatch.

We use chalice framework for building our lambda functions and because of it, we don't have direct access to API-Gateway configuration.

We added below code to cdk_chalice lib `def _update_sam_template(self) `:
```
functions = filter(
    lambda resource: resource['Type'] == 'AWS::Serverless::Api',
    sam_template['Resources'].values()
)
for function in functions:
    function['Properties']['MethodSettings'] = [{
        "DataTraceEnabled": True,
        "HttpMethod": "*",
        "LoggingLevel": "INFO",
        "ResourcePath": "/* "
    }
]
```
and it seemed to work.

Alex, as we agreed, I'm opening the issue, because you wanted to : "tackle it and see whether customizing resources generated by Chalice (and cdk-chalice respectively) can be made easier."

Thank you !
Natasha.

**Comment by alexpulver** [comment]
Thanks for submitting the issue! I'll start by zooming out a bit and recap cdk-chalice intent. The package is meant to automate the following approach to AWS CDK and AWS Chalice interoperability:

1. **Create Chalice application (`chalice new-project`) with default "dev" stage in `.chalice/config.json`.** CDK application could perform this Chalice application scaffolding - create new project or skip this step if project already exists in the target directory (this is `chalice new-project` behavior). I preferred to keep this step manual to hopefully make adoption easier for developers who already have existing Chalice projects

2. **Create CDK application (`cdk init`)**

3. **Use `cdk_chalice.Chalice` class to generate stage per CDK stack in `.chalice/config.json` and run `chalice package`**. This is the main purpose of `cdk-chalice` - to allow passing CDK tokens for resources such as DynamoDB table to SAM template generated by `chalice package` (see example [here](https://github.com/alexpulver/aws-cdk-sam-chalice/blob/master/web-api/.chalice/config.json)), and also to automate the packaging process itself.

Hence if AWS Chalice doesn't support accepting certain options through its configuration mechanism, there are two options I can think of - open an [issue](https://github.com/aws/chalice/issues) for AWS Chalice or customize the resources after they have been imported into the CDK stack by `cdk-chalice`. To make the latter easier, there is a tracking issue in AWS CDK for [migrating CloudFormation templates to the CDK](https://github.com/aws/aws-cdk/issues/7837) - I'd recommend to check this out (it had pull request already submitted and in advanced stages of review)

Does it make sense?

**Comment by alexpulver** [comment]
To clarify, once migrating CloudFormation templates to the CDK is released, I will look for integrating it into cdk-chalice itself, such that you will get CDK objects

**Comment by alexpulver** [comment]
@natCyberark I released [cdk-chalice 0.8.0](https://pypi.org/project/cdk-chalice/0.8.0/) that includes this capability, based on AWS CDK [cloudformation_include](https://docs.aws.amazon.com/cdk/api/latest/docs/cloudformation-include-readme.html) module (it just reached developer preview status). The `cdk-chalice` [API documentation](https://cdk-chalice.softwhat.com/) includes more information and usage example. Would be glad to get your feedback!

**Comment by alexpulver** [comment]
A blog post that walks through that feature: https://softwhat.com/cdk-chalice-adds-support-for-accessing-chalice-generated-cloudformation-resources/
